lib/libpax/libpax.cpp

```diff
@@ -16,14 +16,14 @@ See the License for the specific language governing permissions and
 limitations under the License.
 
 */
-#include "globals.h"
 #include "libpax.h"
+#include "globals.h"
 
 typedef uint32_t bitmap_t;
 enum { BITS_PER_WORD = sizeof(bitmap_t) * CHAR_BIT };
 #define WORD_OFFSET(b) ((b) / BITS_PER_WORD)
 #define BIT_OFFSET(b) ((b) % BITS_PER_WORD)
-#define LIBPAX_MAX_SIZE 0xFFFF  // full enumeration of uint16_t
+#define LIBPAX_MAX_SIZE 0x1FFFF  // full enumeration of uint32_t
```

Comment:
```suggestion
#define LIBPAX_MAX_SIZE 0xFFFF +1  // full enumeration of uint16_t
```
I am not sure what the intention is here. The change does not make the bitmap 32bit as far as i can tell.

But indeed there is a bug. The LIP_MAP_SIZE (next line) is off by due to dividing the max u16 number by bits per word, instead of u16 possible values divided by bits per word.

Reply:
It's intentional to store only the last 2 bytes of MAC addresses, not the full MAC. This allows a small 8Kbyte bitmap organized storage. This way there is no need for a costly find / hash routine to find MAC matches.

Yes, LIBPAX_MAX_SIZE is off by 1. This is a bug. Should be fixed.

Reply:
I tried it, but doesn't work.
`#define LIBPAX_MAX_SIZE 0xFFFF +1  // full enumeration of uint16_t`
leads to
_region `dram0_0_seg' overflowed by 181752 bytes_